src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -438,6 +438,11 @@ private TypeWithAnnotations ComputeType(Binder binder, SyntaxNode syntax, Bindin
                 diagnostics.Add((this.IsIndexer ? ErrorCode.ERR_BadVisIndexerReturn : ErrorCode.ERR_BadVisPropertyType), Location, this, type.Type);
             }
 
+            if (type.Type.IsFileTypeOrUsesFileTypes() && !ContainingType.IsFileTypeOrUsesFileTypes())
```

Comment:
nit: the `IsFileTypeOrUsesFileTypes` on `ContainingType` is a bit odd. Consider splitting the `IsFileTypeOrUsesFileTypes` method into `IsFileTypeOrUsesFileTypes()` and `IsFileType()`. For the check on containing type we only need `IsFileType()`.

Reply:
I find this way convenient as it allows us to treat a type contained in a file type the same as a file type for the purposes of these *signature usage* checks. e.g.

```cs
file class Outer
{
    class Inner
    {
        static void M(Outer outer) { } // ok
    }
}
```